zzzz_modular_occulus/code/modules/mob/living/simple_animal/hostile/siren/replicant.dm

```diff
@@ -0,0 +1,294 @@
+		playsound(src.loc, 'zzzz_modular_occulus/sound/voice/replicanthum.wav', 100, 1, 8, 8)
+		sounddelay = 15
+		return
+	else
+		sounddelay = (sounddelay -1)
+		return
+
+/////////////////Defensive EMP burst starts here///////////////////////
+/mob/living/simple_animal/hostile/siren/replicant/bullet_act()
+	.=..()
+	defensive_burst()
+
+/mob/living/simple_animal/hostile/siren/replicant/attackby()
+	.=..()
+	defensive_burst()
+/mob/living/simple_animal/hostile/siren/replicant/proc/defensive_burst()
+
+	distress_level += 1
+
+	/*
+	In order to make it more likely that players will be around to witness it, lets add more distress if we can
+	see a human player
+
+	*/
+	for (var/mob/living/carbon/human/H in view())
+		if (H.stat != DEAD && H.client)
+			distress_level += 2
+			break
+
+	if (distress_level > 0 && prob(distress_level))
+
+		distress_level = -30 //Once a call is successfully triggered, set the chance negative
+		//So it will be a while before this guy can send another call
+
+		playsound(src.loc, 'zzzz_modular_occulus/sound/voice/roboticactivation.wav', 100, 1, 8, 8)
+		visible_message(SPAN_DANGER("[src] emits a electromagnetic pulse, frying nearby electronics!"))
+		empulse(get_turf(src), emp_range, emp_range, TRUE)
+/////////////////Defensive EMP burst ENDS here///////////////////////
+
+/mob/living/simple_animal/hostile/siren/replicant/death()
+	new /obj/item/weapon/replicant_core(src.loc)
+	..()
+
+/obj/item/weapon/replicant_core
+	name = "replicant core"
+	desc = "All that remains of a creature, it seems to be what remains of it's core. It still seems to glow somewhat."
+	icon = 'icons/obj/food.dmi'
+	icon_state = "boiledrorocore"
+	var/charges = 1
+	var/active = 0
+	var/det_time = 40
+/obj/item/weapon/replicant_core/attack_self(mob/user as mob)
+	activate(user)
+	add_fingerprint(user)
+	if(iscarbon(user))
+		var/mob/living/carbon/C = user
+		C.throw_mode_on()
+	return
+/obj/item/weapon/replicant_core/proc/prime()
+	empulse(src, 4, 10)
+	active = 0
+	return
+/obj/item/weapon/replicant_core/proc/activate(mob/user as mob)
+	if(active)
+		return
+
+	if(user)
+		log_and_message_admins("activates \a [src]")
+		user.attack_log += "\[[time_stamp()]\] <font color='red'>primed \a [src]</font>"
+
+	active = 1
+	playsound(loc, 'sound/weapons/armbomb.ogg', 75, 1, -3)
+
+
+	spawn(det_time)
+		prime(user)
+		return
+
+/mob/living/simple_animal/hostile/siren/replicanttendril
+	name = "replicant tendril"
+	desc = "A thin cord-like tendril made of bio-synthetic mesh, broken off from a larger creature. There are stories of these cords pulling crew into the darkness to never be seen again..."
+	icon = 'zzzz_modular_occulus/icons/mob/siren/replicant.dmi'
+	icon_state = "Replicanttendril"
+	icon_living = "Replicanttendril"
+	icon_aggro = "Replicanttendril"
+	icon_dead = "Replicanttendrildead"
+	mouse_opacity = 2
+	move_to_delay = 0
+	friendly = "buzzes near"
+	vision_range = 10
+	speed = 3
+	maxHealth = 35
+	health = 35
+	melee_damage_lower = 10
```

Comment:
I worry about unarmored people on the ship during these events. Might drop this to 10-15 from 10-18

Reply:
Can do!